src/main/java/ui/MenuControl.java

```diff
@@ -114,27 +152,31 @@ private void onBoardSaveAs() {
         logger.info("Menu: Boards > Save as");
 
         List<PanelInfo> panelList = panels.getCurrentPanelInfos();
-
-        if (panelList.isEmpty()) {
-            logger.info("Did not save new board");
-            return;
-        }
-
         BoardNameDialog dlg = new BoardNameDialog(prefs, mainStage);
         Optional<String> response = dlg.showAndWait();
-        ui.showMainStage();
-        panels.selectFirstPanel();
-        
+
         if (response.isPresent()) {
             String boardName = response.get().trim();
-            prefs.addBoard(boardName, panelList);
-            prefs.setLastOpenBoard(boardName);
-            ui.triggerEvent(new BoardSavedEvent());
-            logger.info("New board " + boardName + " saved");
+            showFirstPanel(panelList);
+            saveCurrentBoard(panelList, boardName);
             ui.updateTitle();
         }
     }
 
+    private void saveCurrentBoard(List<PanelInfo> panelList, String boardName) {
+        prefs.addBoard(boardName, panelList);
+        prefs.setLastOpenBoard(boardName);
+        ui.triggerEvent(new BoardSavedEvent());
+        logger.info("New board " + boardName + " saved");
+    }
+
+    private void showFirstPanel(List<PanelInfo> panelList) {
+        ui.showMainStage();
+        if (!panelList.isEmpty()) {
+            panels.selectFirstPanel();
```

Comment:
This line is to show up the first panel. It was added in [this commit](https://github.com/HubTurbo/HubTurbo/commit/ca91dbb84d91d0e735423cc560476643f08b544c#diff-c522a964929f2e9b705fba9068c9cf9dR212).